### ISSUE TITLE
Video-tracker on linux : videotrackerCombinedCalibrationFile12.json not found and  VIDIOC_QUERYMENU: Invalid argument

### ISSUE THREAD

**Issue by Olm-e** [OSVR/OSVR-Core#357]
hello,
I'm trying to use the HMD (1.2) on linux machine (mint 17 and debian8) with OSVR-Core master branch, compiling well.

I'm using the osvr_server_config.HDK12ExtendedLandscape.sample.json sample renamed as default,
the camera is lightning on, and accessible by guvcview (with root)

I can access and use IMU orientation correctly in python/blender :D, 
but not the position that stays at zero.

when launching the server I receive these errors in the prompt :

`sudo ./osvr_server 
[sudo] password for user: 
[OSVR Server] Using default config file - pass a filename on the command line to use a different one.
[OSVR Server] Using config file 'osvr_server_config.json'
[OSVR Server] Constructing server as configured...
[OSVR Server] Loading auto-loadable plugins...
[OSVR Server] Loading plugins...
[OSVR Server] 
[OSVR Server] Instantiating configured drivers...
[OSVR] Added device: org_osvr_filter_videoimufusion/HeadFusion
[OSVR] Client context initialized for org.osvr.analysisplugin
[OSVR] Interface initialized for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
[OSVR] Could not resolve source for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
[OSVR] Interface initialized for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
[OSVR] Could not resolve source for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
[OSVR Server] Successes:
[OSVR Server]  - com_osvr_VideoBasedHMDTracker/VideoBasedHMDTracker
[OSVR Server]  - org_osvr_filter_videoimufusion/VideoIMUFusion
[OSVR Server] 
[OSVR Server] 
[OSVR Server] Aliases found and parsed from config file.
[OSVR] ERROR: Could not load an object or display descriptor file specified by: /*
    This configuration supports video (so-called "positional") and IMU fusion tracking,
    in addition to orientation-only tracking, with the OSVR HDK.
    It is configured for RenderManager applications in
    non-direct extended-display landscape mode on HDK 1.0-1.2 optics.
    */
"displays/OSVR_HDK_1_1.json"

[OSVR Server] Using OSVR HDK for display configuration. Did not find an alternate valid 'display' object in config file.
[OSVR] ERROR: Could not load an object or render manager config file specified by: "sample-configs/renderManager.extended.landscape.json"

[OSVR Server] Triggering automatic hardware detection...
[OSVR Server] Registering shutdown handler...
[OSVR Server] Starting server mainloop: OSVR Server is ready to go!
vrpn: Connection request received from 127.0.0.1: 127.0.0.1 42312
[OSVR] Performing hardware auto-detection.
[OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKit0
[OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKitPrediction0
VIDIOC_QUERYMENU: Invalid argument
VIDIOC_QUERYMENU: Invalid argument
VIDIOC_QUERYMENU: Invalid argument
VIDIOC_QUERYMENU: Invalid argument
VIDIOC_QUERYMENU: Invalid argument
VIDIOC_QUERYMENU: Invalid argument
VIDIOC_QUERYMENU: Invalid argument
Opening camera 0
[OSVR] Added device: com_osvr_VideoBasedHMDTracker/TrackedCamera0_0
Video-based tracker: NOTE: Beacon calibration filename videotrackerCombinedCalibrationFile12.json was specified, but not found or could not be loaded.
[OSVR] Path tree updated or connection detected
[OSVR] Sending path tree to clients.
[OSVR] Got updated path tree, processing
[OSVR] Constructed a TrackerHandler for com_osvr_Multiserver/OSVRHackerDevKitPrediction0@localhost sensor 0
[OSVR] Successfully produced handler for /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
[OSVR] Constructed a TrackerHandler for com_osvr_VideoBasedHMDTracker/TrackedCamera0_0@localhost sensor 0
[OSVR] Successfully produced handler for /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
[OSVR] Connected 2 of 2 unconnected paths successfully
[OSVR] Performing hardware auto-detection.
[OSVR] Path tree updated or connection detected
[OSVR] Sending path tree to clients.
`

I indeed cannot find the calibration .json file, 

is the problem related to https://github.com/OSVR/OSVR-Core/issues/339 and simple basic lack of video support for uvc cam ?

thanks

**Comment by VRguy** [comment]
A camera firmware upgrade is recommended to access the camera from
OSX/Linux configurations. You can find the upgrade tool here:
http://osvr.github.io/using/

This may not be the only problem, but please start with the upgrade.



On Tue, Feb 16, 2016 at 9:00 AM, Olm-e <notifications@github.com> wrote:

> hello,
> I'm trying to use the HMD (1.2) on linux machine (mint 17 and debian8)
> with OSVR-Core master branch, compiling well.
>
> I'm using the osvr_server_config.HDK12ExtendedLandscape.sample.json sample
> renamed as default,
> the camera is lightning on, and accessible by guvcview (with root)
>
> I can access and use IMU orientation correctly in python/blender :D,
> but not the position that stays at zero.
>
> when launching the server I receive these errors in the prompt :
>
> `sudo ./osvr_server
> [sudo] password for user:
> [OSVR Server] Using default config file - pass a filename on the command
> line to use a different one.
> [OSVR Server] Using config file 'osvr_server_config.json'
> [OSVR Server] Constructing server as configured...
> [OSVR Server] Loading auto-loadable plugins...
> [OSVR Server] Loading plugins...
> [OSVR Server]
> [OSVR Server] Instantiating configured drivers...
> [OSVR] Added device: org_osvr_filter_videoimufusion/HeadFusion
> [OSVR] Client context initialized for org.osvr.analysisplugin
> [OSVR] Interface initialized for
> /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
> [OSVR] Could not resolve source for
> /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
> [OSVR] Interface initialized for
> /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
> [OSVR] Could not resolve source for
> /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
> [OSVR Server] Successes:
> [OSVR Server] - com_osvr_VideoBasedHMDTracker/VideoBasedHMDTracker
> [OSVR Server] - org_osvr_filter_videoimufusion/VideoIMUFusion
> [OSVR Server]
> [OSVR Server]
> [OSVR Server] Aliases found and parsed from config file.
> [OSVR] ERROR: Could not load an object or display descriptor file
> specified by: /*
> This configuration supports video (so-called "positional") and IMU fusion
> tracking,
> in addition to orientation-only tracking, with the OSVR HDK.
> It is configured for RenderManager applications in
> non-direct extended-display landscape mode on HDK 1.0-1.2 optics.
> */
> "displays/OSVR_HDK_1_1.json"
>
> [OSVR Server] Using OSVR HDK for display configuration. Did not find an
> alternate valid 'display' object in config file.
> [OSVR] ERROR: Could not load an object or render manager config file
> specified by: "sample-configs/renderManager.extended.landscape.json"
>
> [OSVR Server] Triggering automatic hardware detection...
> [OSVR Server] Registering shutdown handler...
> [OSVR Server] Starting server mainloop: OSVR Server is ready to go!
> vrpn: Connection request received from 127.0.0.1: 127.0.0.1 42312
> [OSVR] Performing hardware auto-detection.
> [OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKit0
> [OSVR] Added device: com_osvr_Multiserver/OSVRHackerDevKitPrediction0
> VIDIOC_QUERYMENU: Invalid argument
> VIDIOC_QUERYMENU: Invalid argument
> VIDIOC_QUERYMENU: Invalid argument
> VIDIOC_QUERYMENU: Invalid argument
> VIDIOC_QUERYMENU: Invalid argument
> VIDIOC_QUERYMENU: Invalid argument
> VIDIOC_QUERYMENU: Invalid argument
> Opening camera 0
> [OSVR] Added device: com_osvr_VideoBasedHMDTracker/TrackedCamera0_0
> Video-based tracker: NOTE: Beacon calibration filename
> videotrackerCombinedCalibrationFile12.json was specified, but not found or
> could not be loaded.
> [OSVR] Path tree updated or connection detected
> [OSVR] Sending path tree to clients.
> [OSVR] Got updated path tree, processing
> [OSVR] Constructed a TrackerHandler for
> com_osvr_Multiserver/OSVRHackerDevKitPrediction0@localhost sensor 0
> [OSVR] Successfully produced handler for
> /com_osvr_Multiserver/OSVRHackerDevKitPrediction0/semantic/hmd
> [OSVR] Constructed a TrackerHandler for
> com_osvr_VideoBasedHMDTracker/TrackedCamera0_0@localhost sensor 0
> [OSVR] Successfully produced handler for
> /com_osvr_VideoBasedHMDTracker/TrackedCamera0_0/semantic/hmd/front
> [OSVR] Connected 2 of 2 unconnected paths successfully
> [OSVR] Performing hardware auto-detection.
> [OSVR] Path tree updated or connection detected
> [OSVR] Sending path tree to clients.
> `
>
> I indeed cannot find the calibration .json file,
>
> is the problem related to #339
> <https://github.com/OSVR/OSVR-Core/issues/339> and simple basic lack of
> video support for uvc cam ?
>
> thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OSVR/OSVR-Core/issues/357>.
>



-- 
Yuval Boger
CEO | VRguy <http://www.vrguy.net>
Sensics, Inc. <http://www.sensics.com>

Latest news and blog posts:
16 Feb: Vision Summit 2016: Using OSVR to Support (practically) Any Device
in AR/VR
<http://vrguy.blogspot.com/2016/02/vision-summit-2016-using-osvr-to.html>
25 Jan: Got new or custom HMD? Need direct render, time warping, distortion
correction and game engine integration?
<http://vrguy.blogspot.com/2016/01/got-new-hmd-need-direct-render-time.html>
23 Jan: Snow World <http://vrguy.blogspot.com/2016/01/snow-world.html>

**Comment by rpavlik** [comment]
Yes, start with the upgrade. It probably still won't work (well) after that because OpenCV doesn't let us specify a particular camera to use, etc, but it will be a lot closer to working.  Not finding the calibration file is not a fatal error: autocalibration proceeds during usage.  See https://github.com/OSVR/OSVR-Docs/blob/master/Getting-Started/HDK/Video-Based-Tracking-Calibration.md

**Comment by uberscientist** [comment]
Is this the camera upgrade tool you're talking about? http://resource.osvr.com/public_download/Drivers/OSVR_IR_Camera_Firmware_update-5SF006N2_v0007.rar

It's a windows binary and OP and I are on linux, or was there a link I missed?

**Comment by rpavlik** [comment]
Yes, that's the one. Unfortunately there isn't a Linux version available of that firmware update, so you'll have to find/borrow a Windows PC to do the upgrade.

**Comment by Olm-e** [comment]
ok... I also had to find a Windows machine, as it doesn't work on Wine neither ... :/
So the cam is recognised now, and try to callibrate. I had it (badly) working once, but now it stay stuck after the "that distance looks good, hold it right there." sentence in the prompt.
will try to upgrade OSVR-Core with latest changes ...

**Comment by Conzar** [comment]
I just want to update and let you know that updating the firmware does indeed work on linux (I have verified this).  See:
https://github.com/OSVR/OSVR-Docs/blob/master/Utilities/UpdateHDKLinux.md

**Comment by rpavlik** [comment]
@conzar the IR camera has its own firmware described above, and that update does not run on Linux.
